### PR TITLE
TH2-2345. Ability to receive the first message index in the message stream

### PR DESCRIPTION
**Pull request by MrJonniK**

### CODE REVIEW

File: cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java [hunk 1]
```diff
@@ -479,7 +513,7 @@ public final StoredMessageId getNearestMessageId(String streamName, Direction di
 		StoredMessageId result = doGetNearestMessageId(streamName, direction, timestamp, timeRelation);
 		logger.debug("First message ID appeared on {} or {} for stream '{}' and direction '{}' got", 
 				timestamp, timeRelation.getLabel(), streamName, direction.getLabel());
-		return result;
+		return result;
```
Comment (Vladimir-Panarin-Exactpro) [inline cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java:516]: Extra space at line end. Remove it.

File: cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/MessageBatchOperator.java [hunk 2]
```diff
@@ -70,10 +70,16 @@ DetailedMessageBatchEntity get(UUID instanceId, String streamName, String direct
 	@Select
 	PagingIterable<DetailedMessageBatchEntity> getAll(Function<BoundStatementBuilder, BoundStatementBuilder> attributes);
 	
+	@Query("SELECT * FROM ${qualifiedTableId} WHERE "
+			+INSTANCE_ID+"=:instanceId AND "+STREAM_NAME+"=:streamName AND "+DIRECTION+"=:direction LIMIT 1")
+	DetailedMessageBatchEntity getFirstIndex(UUID instanceId, String streamName, String direction,
```
Comment (Vladimir-Panarin-Exactpro) [inline cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/MessageBatchOperator.java:75]: For this particular query you don't need to get the whole message batch, only message_index value as Long.

**Review by Vladimir-Panarin-Exactpro — CHANGES_REQUESTED** [review]

File: cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/MessageBatchOperator.java (hunk 2 above)
Comment (MrJonniK) [reply cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/MessageBatchOperator.java:75]: The method with long or Long return type will fail if the result set is empty.

Comment (MrJonniK) [reply cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/MessageBatchOperator.java:75]: But I can use Row type without mapping

File: gradle.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
-release_version = 2.19.0
+release_version = 2.19.1
```
Comment (Vladimir-Panarin-Exactpro) [inline gradle.properties:1]: No need to increase version number as this is not merged to master yet.

**Review by Vladimir-Panarin-Exactpro — APPROVED** [review]

**Pull request merged**